### DISCUSSION TITLE
Support and Large-Scale Deployment Experience with Odigos on VMs

### DISCUSSION THREAD

**Discussion by ANSANJAY** [odigos-io/odigos#938] [Q&A]
Hello Odigos Community,

I'm exploring Odigos for observability in our infrastructure and have a few questions regarding its capabilities and real-world usage:

1. **Support for VMs**: Does Odigos support monitoring applications running on Virtual Machines (VMs)? If so, what level of support is offered out of the box, and how customizable is it for VM-based environments?

2. **Kubernetes (K8s) Integration**: We understand that Odigos supports Kubernetes environments. Can you share more about the extent of this support and any specific features or integrations available for K8s?

3. **Real-World Usage**: Are there any organizations that have been using Odigos for large-scale deployments on VMs? We would love to hear about any case studies, experiences, or insights from deployments at scale.

Thank you for your help and looking forward to your insights!

Best regards,
Anamika

**Comment by edeNFed** [comment]
Hi @ANSANJAY thanks for reaching out. Please take a look at the answers below.

1. **VM support** - yes this is absolutely possible. We have a beta version of the VM support and multiple companies are already using it. I would be happy to give you access to it for testing.

2. **Kubernetes integrations** - there are many integrations that Odigos offers for Kubernetes. A few examples:
* All the OpenTelemetry data we produce is enriched with [Kubernetes resources according to the semantic conventions](https://opentelemetry.io/docs/specs/semconv/resource/k8s/)
* Infrastructure metrics for nodes/pods/containers

There are more examples, if you can point me to what interests you I can elaborate more

3. **Real world usage** - Most of the companies that use Odigos today are large enterprises. We have users from multiple sectors including banks, and e-commerce, that are running Odigos in production. Some of those users are extremely large enterprises (Fortune 100) that use Odigos on 10,000s of microservices.

From my experience, the two things that attract these large companies the most are:
- Effortless - instrument all your microservices with a click of a button, without requiring developers to do code changes. This enables the platform team to offer best-in-class observability as a service for the development teams.
- Performance overhead - as our instrumentation is based on eBPF the performance overhead is extremely low (less than 1% in our testing). This is especially useful for Java/Python/NodeJS shops where other (non-eBPF) automatic instrumentation solutions can cause severe performance overhead.

Hope that helps. Please let me know if you have any more questions.

**Reply by Arirec** [reply]
Hi Anamika, hope all is well.
Just wanted to follow up and see if you had a chance to discuss Odigos.
Happy to help or assist. Let us know
Thanks
Ari
